packages/http/libraries/core/src/http/adapter/InversifyHttpAdapter.ts

```diff
@@ -212,12 +226,24 @@ export abstract class InversifyHttpAdapter<
         }
       } catch (error: unknown) {
         this.#printError(error);
-        return this.#reply(req, res, new InternalServerErrorHttpResponse());
+
+        if (
+          InversifyHttpAdapterError.isErrorOfKind(
+            error,
+            InversifyHttpAdapterErrorKind.pipeError,
+          )
+        ) {
+          return this.#reply(req, res, new BadRequestHttpResponse());
```

Comment:
The user might want to fail with a different http error code. 401, 403, 404, 409, 422 might be possible codes as well in some use cases.

We would need a way to allow the user to throw a custom error code.

Reply:
I added a simple approach